client/src/modules/journal/journal.js

```diff
@@ -512,6 +539,9 @@ function JournalController(Journal, Sorting, Grouping,
   function editTransactionType(row) {
```

Comment:
We should eventually make this a `ui-grid` dropdown editor - it would remove some code, and you wouldn't have to call `propogate` (because it would be caught by the `afterCellEdit` method you created).

I'll make an issue about it - no need to address it in the PR.